### PR TITLE
Disallow using 'context' in tests

### PR DESCRIPTION
**Pull request by kimjoar**
See https://github.com/elastic/kibana/pull/10991

### CODE REVIEW

**Comment by kimjoar** [comment]
If people are +1 on this I can create a PR for xpack too.

File: .eslintrc.yaml [hunk 1]
```diff
@@ -52,6 +52,7 @@ rules:
   no-path-concat: off
   no-proto: error
   no-redeclare: error
+  no-restricted-globals: [ error, context ]
```
Comment (spalger) [inline .eslintrc.yaml:55]: I'd love to add things like `error`, `name`, but it's probably best to keep it simple for this change.

**Review by spalger — APPROVED** [review]

File: .eslintrc.yaml (hunk 1 above)
Comment (kimjoar) [reply .eslintrc.yaml:55]: Not sure I follow, @spalger. Are you thinking of restricting other globals, or of writing this rule differently?

Comment (spalger) [reply .eslintrc.yaml:55]: restricting other globals, just complaining about `window.event` and `window.name`, among others...

Comment (kimjoar) [reply .eslintrc.yaml:55]: ++, good point. It's def something we should consider.

**Pull request merged**